tests/unit/executors/encoders/numeric/test_transform_encoder.py

```diff
@@ -1,20 +1,26 @@
 import os
 import pickle
-import shutil
 
+import pytest
 import numpy as np
 
+from jina.executors.metas import get_default_metas
 from jina.executors.encoders.numeric import TransformEncoder
-from tests import rm_files
 
 input_dim = 28
 target_output_dim = 2
 
+@pytest.fixture(scope='function', autouse=True)
+def metas(request, tmpdir):
+    os.environ['TEST_WORKSPACE'] = str(tmpdir)
```

Comment:
I'd prefix it with `JINA_`, so it becomes `JINA_TEST_WORKSPACE` to guarantee, that there are no namespace collisions.

Furthermore, having the same environment variable for each and every test, might be a bad idea. What about:

```
@pytest.fixture()
def random_workspace_name():
    return f'JINA_TEST_WORKSPACE_{random.rand()}'
    # just make the random better...

@pytest.fixture(scope='function', autouse=True)
def metas(request, tmpdir, random_workspace_name):
    os.environ[random_workspace_name] = str(tmpdir)
    ...
```

This allows to explicitly give a workspace name in a test function:

```
@pytest.mark.parametrize('random_workspace_name', ['JINA_TEST_WORKSPACE_EXPLICIT_NAME'])
def my_test():
    assert metas['workspace'] == os.environ['JINA_TEST_WORKSPACE_EXPLICIT_NAME']
    # TRUE
```

Reply:
randomizing may be problematic for tests where you need to set the workspace inside a `yml` file?

Reply:
I accidentally deleted @JoanFM comment, that this might be a bad idea, since `.yml` files might need and explicit name.

But anyhow, then you can explicitly set it as shown in the `my_test` function.

Reply:
great suggestions! I applied most of the suggestions into this PR, please take a look :) @maximilianwerk